doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -126,42 +133,72 @@ Type declarations
+      tac2alg_constructor ::= @ident
+      | @ident ( {*, @tac2type5 } )
+      tac2rec_field ::= {? mutable } @ident : @tac2type5
+
+   :n:`:=`
+     Defines a type with a manifest
+
+   :n:`::=`
+     Defines an open variant type, a special kind of variant type whose constructors are not
+     statically defined, but can instead be extended dynamically. A typical example
+     is the standard `exn` type for exceptions. Pattern matching on open variants must always
+     include a catch-all clause. They can be extended with this form, in which case
+     :token:`tac2typ_knd` should be in the form :n:`[ {? {? %| } {+| @tac2alg_constructor } } ]`.
+
+     .. todo PR above: does it add an additional constructor for an existing o v type or
+        does it create both the type (and apparently the constructor?
+
+   Without :n:`{| := | ::= }`
+     Defines an abstract type for use representing data from OCaml.  Not for
+     end users.
+
+   Each production of :token:`tac2typ_knd` defines one of four possible kinds
+   of definitions, respectively: alias, variant, open variant and record types.
+
+   .. cleanup
+       .. productionlist:: coq
+          ltac2_typedef    : `ltac2_type`
+                           : [ `ltac2_constructordef` | ... | `ltac2_constructordef` ]
+                           : { `ltac2_fielddef` ; ... ; `ltac2_fielddef` }
+                           : [ .. ]
+          ltac2_constructordef   : `uident` [ ( `ltac2_type` , ... , `ltac2_type` ) ]
+          ltac2_fielddef         : [ mutable ] `ident` : `ltac2_type`
+
+   Aliases are names for a given type expression and are transparently
    unfoldable to it. They cannot be recursive. The non-terminal
-   :production:`uident` designates identifiers starting with an uppercase.
+   :token:`qualid` designates identifiers starting with an uppercase.
 
    Variants are sum types defined by constructors and eliminated by
    pattern-matching. They can be recursive, but the `rec` flag must be
    explicitly set. Pattern matching must be exhaustive.
 
+   .. todo PR next sentence: is that right?
+
+   Open variants can be extended with additional constructors using the `::=` form.
+
    Records are product types with named fields and eliminated by projection.
    Likewise they can be recursive if the `rec` flag is set.
 
-   .. cmdv:: Ltac2 Type {? @ltac2_typeparams } @ltac2_qualid ::= [ @ltac2_constructordef ]
+.. cmd:: Ltac2 @ external @ident : @tac2type5 := @string @string
+   :name: Ltac2 external
 
-      Open variants are a special kind of variant types whose constructors are not
-      statically defined, but can instead be extended dynamically. A typical example
-      is the standard `exn` type. Pattern matching on open variants must always include a catch-all
-      clause. They can be extended with this command.
+   This command defines an abstract type. It has no use for the end user and
+   is dedicated to types representing data coming from the OCaml world.
```

Comment:
From what I can see in the code `Ltac2 @ external` is always used to define abstract Ltac2 terms (generally functions) and never Ltac2 types (which can already be declared abstract with `Ltac2 Type` without `:=`).